### DISCUSSION TITLE
Issue while running tests

### DISCUSSION THREAD

**Discussion by shivabhardwaj18** [seleniumbase/SeleniumBase#2803] [Q&A] [answered]
Hello,

I recently upgraded to following (latest) versions:
Python - 3.12.3
Seleniumbase - 4.27.0
pytest - 8.2.0
pytest-bdd - 7.1.2

Suddenly, I started getting following error:

<img width="933" alt="Screenshot 2024-05-24 093857" src="https://github.com/seleniumbase/SeleniumBase/assets/76955043/ebb09389-dfcc-4f3c-a376-eb5ce645b579">

I think latest versions are not compatible with each other but I am not too sure. Can someone please help?

Thanks in advance,
Shiva

**Comment by mdmintz** [comment, marked as answer]
That's a `pytest-bdd` issue. Not a `seleniumbase` one. If you want BDD with SeleniumBase, use the built-in `behave` integration: [SeleniumBase/examples/behave_bdd](https://github.com/seleniumbase/SeleniumBase/tree/master/examples/behave_bdd)

**Reply by shivabhardwaj18** [reply]
Hello @mdmintz,

We have been using pytest-bdd and seleniumbase from quite a long time. I think it will be too difficult for me to use behave instead of pytest-bdd as all our tests currently use pytest-bdd. Do you have any suggestions?

Thanks in advance,
Shiva

**Reply by mdmintz** [reply]
`pytest-bdd` released a new version on June 4th (2024): https://pypi.org/project/pytest-bdd/7.2.0/
(If there's an issue there, that's a separate framework.)
